arangod/Aql/AllRowsFetcher.cpp

```diff
@@ -26,21 +26,35 @@
 #include "Aql/AqlItemMatrix.h"
 #include "Aql/DependencyProxy.h"
 #include "Aql/InputAqlItemRow.h"
+#include "Aql/ShadowAqlItemRow.h"
 
 using namespace arangodb;
 using namespace arangodb::aql;
 
```

Comment:
Since we were discussing it this morning: Either in the header file or this implementation file itwould be helpful to have a paragraph that describes what `AllRowsFetcher` does, with some detail wrt what happens if upstream is in state `WAITING`, and what happens with `ShadowRow`s (as they seem to be adding some complications)

Reply:
Agree!